ext/rugged/rugged_patch.c

```diff
@@ -344,6 +353,18 @@ static int patch_print_cb(
 
 	rb_ary_push(rb_buffer, rb_str_new(line->content, line->content_len));
 
+	if (print_payload->timeout > 0) {
+		struct timeval finish;
+		gettimeofday(&finish, NULL);
```

Comment:
@vmg we can probably just `#ifdef` this but `clock_gettime` doesn't exist on OSX until Sierra.

For now we could either stick with `gettimeofday` on OSX or `mach_absolute_time`? It may not work exactly the same as `clock_gettime`, but I think it's monotonic?